[CI] MeterRegistryConcurrencyTests testLockingWhenRegistering failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/3dxgybhdqdrbq/tests/:modules:apm:test/org.elasticsearch.telemetry.apm.MeterRegistryConcurrencyTests/testLockingWhenRegistering
**Reproduction line:**
```
./gradlew ':modules:apm:test' --tests "org.elasticsearch.telemetry.apm.MeterRegistryConcurrencyTests.testLockingWhenRegistering" -Dtests.seed=AA78182D0641F5E8 -Dtests.locale=ar-LY -Dtests.timezone=Atlantic/Cape_Verde -Druntime.java=21
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.telemetry.apm.MeterRegistryConcurrencyTests&tests.test=testLockingWhenRegistering


**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: sameInstance(<org.elasticsearch.telemetry.apm.MeterRegistryConcurrencyTests$1@6ef477b8>)
     but: was <io.opentelemetry.api.metrics.DefaultMeter@5b2f197a>

  at __randomizedtesting.SeedInfo.seed([AA78182D0641F5E8:945C1E5F0B2CB55E]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.telemetry.apm.MeterRegistryConcurrencyTests.lambda$testLockingWhenRegistering$4(MeterRegistryConcurrencyTests.java:113)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1209)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1182)
  at org.elasticsearch.telemetry.apm.MeterRegistryConcurrencyTests.testLockingWhenRegistering(MeterRegistryConcurrencyTests.java:113)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
  at java.lang.reflect.Method.invoke(Method.java:580)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:1583)

```

--- a/modules/apm/src/test/java/org/elasticsearch/telemetry/apm/MeterRegistryConcurrencyTests.java
+++ b/modules/apm/src/test/java/org/elasticsearch/telemetry/apm/MeterRegistryConcurrencyTests.java
@@ -90,7 +90,6 @@ public class MeterRegistryConcurrencyTests extends ESTestCase {
         }
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/101725")
     public void testLockingWhenRegistering() throws Exception {
         APMMeterRegistry meterRegistrar = new APMMeterRegistry(lockingMeter);
 
@@ -106,12 +105,14 @@ public class MeterRegistryConcurrencyTests extends ESTestCase {
         // assert that a thread is waiting for a lock during long-running registration
         assertBusy(() -> assertThat(setProviderThread.getState(), equalTo(Thread.State.WAITING)));
         // assert that the old lockingMeter is still in place
-        assertBusy(() -> assertThat(meterRegistrar.getMeter(), sameInstance(lockingMeter)));
+        assertThat(meterRegistrar.getMeter(), sameInstance(lockingMeter));
 
         // finish long-running registration
         registerLatch.countDown();
+        // wait for everything to quiesce, registerLatch.countDown() doesn't ensure lock has been released
+        assertBusy(setProviderThread::join);
+        assertBusy(registerThread::join);
         // assert that a meter was overriden
-        assertBusy(() -> assertThat(meterRegistrar.getMeter(), sameInstance(lockingMeter)));
-
+        assertThat(meterRegistrar.getMeter(), sameInstance(noopMeter));
     }
 }

--- a/modules/apm/src/test/java/org/elasticsearch/telemetry/apm/MeterRegistryConcurrencyTests.java
+++ b/modules/apm/src/test/java/org/elasticsearch/telemetry/apm/MeterRegistryConcurrencyTests.java
@@ -110,8 +110,8 @@ public class MeterRegistryConcurrencyTests extends ESTestCase {
         // finish long-running registration
         registerLatch.countDown();
         // wait for everything to quiesce, registerLatch.countDown() doesn't ensure lock has been released
-        assertBusy(setProviderThread::join);
-        assertBusy(registerThread::join);
+        setProviderThread.join();
+        registerThread.join();
         // assert that a meter was overriden
         assertThat(meterRegistrar.getMeter(), sameInstance(noopMeter));
     }

